dulwich/tests/compat/test_repository.py

```diff
@@ -179,13 +181,21 @@ def test_git_worktree_list(self):
         worktrees = self._parse_worktree_list(output)
         self.assertEqual(len(worktrees), self._number_of_working_tree)
         self.assertEqual(worktrees[0][1], '(bare)')
-        self.assertEqual(worktrees[0][0], self._mainworktree_repo.path)
+        if sys.platform == 'win32':
+            # on windows, paths are case insensitive, and native git reports paths with '/' as seperator
```

Comment:
Please use os.path.normcase here - that should remove the need for Windows-specific code. normcase is a no-op on POSIX.